src/gt4py/backend/base.py

```diff
@@ -320,15 +320,11 @@ def make_args_data_from_iir(implementation_ir: gt_ir.StencilImplementation) -> D
                         dtype=implementation_ir.fields[arg.name].data_type.dtype,
                         boundary=implementation_ir.fields_extents[arg.name].to_boundary(),
                     )
-                else:
```

Comment:
May be redundant to keep unused fields both in `field_info` and `unreferenced` dictionaries. Same applies to unused parameters. So I suggest to keep unused arguments in `unreferenced` only.

Reply:
I see your point here, but this change could be potentially API changing, if someone expects to find information for all the fields (even the unused ones) inside `field_info`, so please make this change explicit in the PR description.

Reply:
Ok, after thinking again about this, I now see some compatibility problems with this change. Right now all the backends assume they get all the fields, even if they are not used (even the dawn backends, which need to convert the None to something else). It's better for now leave this code untouched.